src/ProjectService.Validation/CreateProjectValidator.cs

```diff
@@ -1,91 +1,179 @@
-            {
-                RuleFor(project => project.ShortDescription)
-                    .MaximumLength(300)
-                    .WithMessage("Project short description is too long.");
-            });
+      When(project => project.Users != null && project.Users.Any(), () =>
+      {
+        RuleForEach(project => project.Users).ChildRules(user =>
+          {
+            user.RuleFor(user => user.UserId)
+              .NotEmpty();
+
+            user.RuleFor(user => user.Role)
+              .IsInEnum();
+          });
+      });
 
-            When(project => !string.IsNullOrEmpty(project.Description?.Trim()), () =>
+      When(project => project.ProjectImages != null && project.ProjectImages.Any(), () =>
+      {
+        RuleForEach(project => project.ProjectImages)
+          .Must(x => !string.IsNullOrEmpty(x.Content))
+          .WithMessage("Content can't be empty")
+          .Must(x => imageFormats.Contains(x.Extension))
+          .WithMessage("Wrong extension")
+          .Must(images => images.Name.Length < 150)
+          .WithMessage("Name's length must be less than 150 letters")
+          .Must(images =>
             {
-                RuleFor(project => project.Description)
-                    .MaximumLength(300)
-                    .WithMessage("Project description is too long.");
-            });
-
-            When(
-                news => news.DepartmentId.HasValue,
-                () =>
-                    RuleFor(news => news.DepartmentId)
-                        .Must(DepartmentId => DepartmentId != Guid.Empty)
-                        .WithMessage("Wrong type of department Id."));
+              try
+              {
+                var byteString = new Span<byte>(new byte[images.Content.Length]);
+                return Convert.TryFromBase64String(images.Content, byteString, out _);
+              }
+              catch
+              {
+                return false;
+              }
+            }).WithMessage("Wrong image content.");
+      });
+
+      When(project => !string.IsNullOrEmpty(project.ShortDescription?.Trim()), () =>
+      {
+        RuleFor(project => project.ShortDescription)
+          .MaximumLength(300)
+          .WithMessage("Project short description is too long.");
+      });
+
+      When(project => !string.IsNullOrEmpty(project.Description?.Trim()), () =>
+      {
+        RuleFor(project => project.Description)
+          .MaximumLength(300)
+          .WithMessage("Project description is too long.");
+      });
+
+      When(
+        news => news.DepartmentId.HasValue,
+        () =>
+          RuleFor(news => news.DepartmentId)
+            .Must(DepartmentId => DepartmentId != Guid.Empty)
+            .WithMessage("Wrong type of department Id."));
+    }
+
+    private bool CheckDepartmentExistence(Guid? departmentId)
+    {
+      if (!departmentId.HasValue)
+      {
+        return true;
```

Comment:
if departmentId is null - why we return true? that means that Department with id null is exist

Reply:
We return the trueas, because the meaning of the method is to check that the id of a nonexistent department is not specified.

Reply:
done